python-package/xgboost/spark/core.py

```diff
@@ -198,11 +197,16 @@ class _SparkXGBParams(
         "The number of XGBoost workers. Each XGBoost worker corresponds to one spark task.",
         TypeConverters.toInt,
     )
+    device = Param(
+        Params._dummy(),
+        "device",
+        "The device type for XGBoost workers. Available options are cpu,cuda and gpu.",
+        TypeConverters.toString,
+    )
     use_gpu = Param(
         Params._dummy(),
         "use_gpu",
-        "A boolean variable. Set use_gpu=true if the executors "
-        + "are running on GPU instances. Currently, only one GPU per task is supported.",
+        "Deprecated, use `device` instead.",
```

Comment:
Let's keep original doc but append Deprecated

Reply:
Done, the notice is prepended.